src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,429 @@
+---
+title: "Codelab: Basic Flutter layout concepts"
+description: "A codelab that teaches basic Flutter layout concepts through DartPad2 code examples."
+toc: true
+--- 
+{{site.alert.important}}
+  This codelab covers basic Flutter layout concepts using a code editor called DartPad2. An understanding of [Dart](https://www.dart.dev) or some exposure to object-oriented programming is recommended. 
+{{site.alert.end}}
+{{site.alert.important}}
+  Currently, DartPad2 is in the experimental stage and being tested with Google Chrome. A PC (Microsoft, Mac, or Linux) with Google Chrome installed is required for this codelab. If you encounter any difficulties while using DartPad2, please create a [DartPad issue](https://github.com/dart-lang/dart-pad/issues/new) and label the issue "platform-web."
+{{site.alert.end}} 
+{{site.alert.note}}
+  Flutter supports devices that maintain right-to-left language settings. 
+{{site.alert.end}}
+## Introduction
+
+In this codelab, you'll interact with and build Flutter layouts in a code editor called DartPad2. 
+As you progress through this codelab, you learn that widgets build everything in Flutter. 
+A widget is an immutable object that describes a specific part of a UI. You also learn that, 
+while widgets are simple in their functions, widgets can be complex in their structures. 
+Widgets can be built around and inside of other widgets, 
```

Comment:
I this we should talk about this a bit more because I think this makes sense as you work with widgets like Flexible, Expanded, and SizedBox. Plus this terminology is present in Andrew's original document.